### ISSUE TITLE
Every properties of embedded entity has value `undefined` when loaded twice

### ISSUE THREAD

**Issue by mung9** [mikro-orm/mikro-orm#3134]
**Describe the bug**
If an embedded entity loaded twice inside a UoW and the entity got updated inside the other UoW between the two load, loaded properties are filled with `undefined` except that got updated

**To Reproduce**
```ts
@Embeddable()
class Nested {

  @Property()
  field1: string;

  @Property()
  field2: string;

  @Property()
  field3: string;

  constructor(field1: string, field2: string, field3: string) {
    this.field1 = field1;
    this.field2 = field2;
    this.field3 = field3;
  }

}

@Entity()
class Parent {

  @PrimaryKey({ autoincrement: false })
  id: number;

  @Embedded(() => Nested)
  nested: Nested;

  constructor(id: number, nested: Nested) {
    this.id = id;
    this.nested = nested;
  }

}

// initialize
const nested = new Nested('A', 'B', 'C');
await orm.em.fork().persistAndFlush(new Parent(1, nested));

// first load
const em1 = orm.em.fork();
const p1 = await em1.findOneOrFail(Parent, 1);
expect(p1.nested.field1).toBe('A');
expect(p1.nested.field2).toBe('B');
expect(p1.nested.field3).toBe('C');

// update inside other UoW
const em2 = orm.em.fork();
const p = await em2.findOneOrFail(Parent, 1);
p.nested.field1 = 'Z';
await em2.persistAndFlush(p);

// reload
const p2 = await em1.findOneOrFail(Parent, 1);

console.log(p2.nested) // => `{ field1: 'Z' }`
```

**Expected behavior**
Expect `field2` and `field3` to be `B` and `C` respectively, but both are `undefined`

**Versions**

| Dependency            | Version |
| --------------------- | ------- |
| node                  | 16.13.1 |
| typescript            | 4.6.4   |
| @mikro-orm/core       | 5.1.4   |
| @mikro-orm/postgresql | 5.1.4 |
| driver | pg |

**Comment by mung9** [comment]
**reproduction code**
https://github.com/mung9/mikro-orm/blob/master/tests/features/embeddables/GH3134.test.ts

**Comment by B4nan** [comment]
Interesting, thanks for the repro!

Looks like the data are stored just fine, if you would reload that entity you would see the correct output.

**Comment by mung9** [comment]
I've wrote some code to fix this issue. can you take a look?
https://github.com/mung9/mikro-orm/commit/cd4f57b5344d66bf48047dc82a491b3139b960a3

It create embeddable entity only when exists entity doesn't have embeddable entity

(I'm sorry for my poor english..)

**Comment by mung9** [comment]
`ObjectHydrator` create and assign new embeddable entity every time it hydrates data. 
If it overwrite the entity instaed of replace it, existing data that is not related to the diff can be preserved

https://github.com/mikro-orm/mikro-orm/blob/60ec052cae0130f4dc97bdbbfecc5cccc1504790/packages/core/src/entity/EntityFactory.ts#L95

**Comment by B4nan** [comment]
Fixed by #3135, thanks again!
